src/etc/inc/captiveportal.inc

```diff
@@ -184,15 +185,14 @@ function captiveportal_configure_zone($cpcfg) {
 
 		if (platform_booting()) {
 			echo "Starting captive portal({$cpcfg['zone']})... ";
-
-			/* remove old information */
-			unlink_if_exists("{$g['vardb_path']}/captiveportal{$cpzone}.db");
 		} else {
 			captiveportal_syslog("Reconfiguring captive portal({$cpcfg['zone']}).");
 		}
 
-		/* init ipfw rules */
+		/* (re)init ipfw rules. Cause all users to disconnect */
+		captiveportal_radius_stop_all();
 		captiveportal_init_rules(true);
+		unlink_if_exists("{$g['vardb_path']}/captiveportal{$cpzone}.db");
```

Comment:
For the main CP zone : Because currently when settings are saved, captiveportal rules are re-appplied unconditionally to the network interface, meaning **all ipfw rules are unconditionally flushed**.

This is a big problem when editing captive portal settings while some users are connected : When saving the settings, **users go technically disconnected** and are redirected to the login page (because ipfw rules are flushed),  but they are still considered as connected and **are unable to log-in again**(because they are still present in the sqlite database).

I solved this issue by flushing SQLite DB when rules are re-generated, and i added a warning "there is connected users, they will get disconnected. Do you want to continue ?" on the GUI. 

For the RADIUS database : because this database now obsolete, captiveportal don't use it anymore.

Reply:
> all ipfw rules are unconditionally flushed.

That would be considered a separate issue and would be under its own redmine issue and PR. Though it's tempting to try rolling everything into one, everything should be as narrowly scoped as possible.

Reply:
@jim-p No problem. I'll remove this fix and submit it in another PR+Redmine issue.

Should i remove the [GUI Warning](https://github.com/pfsense/pfsense/pull/3640/files#diff-7e49d642280675092542de8bd5454f3cR1202) as well?

Reply:
Yes, take that out as well and combine it with the other issue. Ideally we shouldn't clear out the users/rules/etc when saving portal settings so updating settings is less disruptive, but if we can't make that work then yes it should be consistent in the other direction.